src/commands/createApiCenterService.ts

```diff
@@ -0,0 +1,95 @@
+        if (!locations || locations.length === 0) {
+            vscode.window.showWarningMessage(UiStrings.NoLocationAvailable);
+            return;
+        }
+        const location = await vscode.window.showQuickPick(locations, { placeHolder: UiStrings.SelectLocation });
+        if (!location) {
+            return;
+        }
+
+        await vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: UiStrings.CreateApiCenterService
+        }, async (progress, token) => {
+            progress.report({ message: UiStrings.GetResourceGroup });
+            const rgExist = await apiCenterService.isResourceGroupExist();
+            if (!rgExist) {
+                progress.report({ message: UiStrings.CreatingResourceGroup });
+                await AzureApiCenterService.confirmResourceGroupWithRetry(apiCenterService, location);
+            }
+
+            progress.report({ message: UiStrings.CreatingApiCenterService });
+            validateResponse(await apiCenterService.createOrUpdateApiCenterService(location));
+            // Retry mechanism to check API center creation status
+            await AzureApiCenterService.confirmServerStatusWithRetry(serverName, node, context);
+        });
+    };
+    export function validateResponse(response: any) {
+        if (response && (response as any).message) {
+            throw new Error((response as any).message);
+        }
+    };
+    export async function confirmResourceGroupWithRetry(apiCenterService: ApiCenterService, location: string) {
+        let retryCount = 0;
+        const maxRetries = 5;
+        const retryDelay = 10000; // 10 seconds
+        let result: ResourceGroup;
+        do {
+            validateResponse(result = await apiCenterService.createOrUpdateResourceGroup(location));
+            if (result && result.properties && result.properties.provisioningState && result.properties.provisioningState === 'Succeeded') {
+                break;
+            }
+            retryCount++;
+            await GeneralUtils.sleep(retryDelay);
+        } while (retryCount < maxRetries);
+        if (retryCount >= maxRetries) {
+            throw new Error(UiStrings.LongTimeToCreateResourceGroup);
+        }
+    };
+    export async function confirmServerStatusWithRetry(serverName: string, node: SubscriptionTreeItem, context: IActionContext) {
+        let retryCount = 0;
+        const maxRetries = 5;
+        const retryDelay = 10000; // 10 seconds
+        do {
+            try {
+                const resourceGraphService = new ResourceGraphService(node.subscription);
+                const apiCenter = await resourceGraphService.queryApiCenterByName(serverName);
+                if (apiCenter && apiCenter.length > 0) {
+                    await node.refresh(context);
+                    vscode.window.showInformationMessage(UiStrings.CreateResourceSuccess);
```

Comment:
Move this line before node.refresh()?

Reply:
put `node.refresh` before `showInfo` is telling finish the created action, and the treeview show the created item. 
Display all results to the user, including the node representations, so that the user can understand that it is a genuine creation action.